### PR TITLE
Release notes for PS 5.7.21-20.ease-notes/release-notes_index.rst

### PR DESCRIPTION
**Pull request by fiowro**
Also includes minimal fixes for DOC-695 (documenting server restart
on detected data corruption) and DOC-704 (version_suffix variable)

### CODE REVIEW

**Review by george-lorch — APPROVED** [review]
TDB and MYR are accurate, but I think I need to do a better job of exposing fixes and improvements to MYR that comes as part of upstream merges so they can be reflected in our release notes.

**Review by hrvojem — CHANGES_REQUESTED** [review]
Please add the PS changes as mentioned on Slack

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+<http://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-21.html>`_
+and includes all the bug fixes in it.
+Percona Server 5.7.21-20 is now the current GA (Generally Available) release
+in the 5.7 series. All software developed by Percona is open-source and free.
+Details of this release can be found in the `5.7.21-20 milestone on Launchpad
+<https://launchpad.net/percona-server/+milestone/5.7.21-20>`_.
+
+New Features
+============
+
```
Comment (percona-ysorokin) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: `version_comment` is no longer read-only. This should be mentioned here as well

File: doc/source/ps-variables.rst [hunk 2]
```diff
@@ -612,6 +612,11 @@ System Variables
      - Yes
      - Global
      - Yes
+   * - :variable:`version_suffix`
+     - Yes
+     - Yes
+     - Global
+     - Yes
```
Comment (percona-ysorokin) [inline doc/source/ps-variables.rst:625]: Most probably `version_comment` should be mentioned here as well as its behavior is not the same as in upstream - it's no longer read-only.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  :mysqlbug:`89421`, and :mysqlbug:`89422`). 
+
+* CMake error took place at compilation with bundled zlib. Bug fixed 
+  :psbug:`302`.
+
+* A GCC 7 warning fix introduced regression in |Percona Server| that lead to
+  a wrong SQL query built to access the remote server when Federated storage
+  engine was used. Bug fixed :psbug:`1134`. 
+
+* It was possible to enable ``encrypt_binlog`` with no binary logging. Bug
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: :variable:`encrypt_binlog` ?

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  fixed :psbug:`287`.
+
+* Long buffer wait times where occuring on busy servers in case of several
+  parallel workers. Bug fixed :psbug:`276`.
+
+* Server queries that contained JSON special characters and were logged by
+  ``audit_log`` plugin in JSON format caused invalid output due to lack of
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: :ref:`audit_log_plugin`

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  escaping. Bug fixed :psbug:`1115`.
+
+* Percona Server now uses *Travis CI*  for additional tests. Bug fixed
+  :psbug:`3777`.
+
+Other bugs fixed: :psbug:`257`, :psbug:`1090` (upstream :mysqlbug:`78048`), :psbug:`1109`, :psbug:`1127`, :psbug:`2415`,
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: doc8

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+* mysqld startup script was unable to detect jemalloc library location for
+  preloading, and that prevented starting |Percona Server| on systemd based
+  machines. Bugs fixed :psbug:`3784` and :psbug:`3791`.
+
+* There was a problem with fulltext search, which could find a word with
+  punctuation marks in natural language mode only, but not in boolean mode.
+  Bugs fixed :psbug:`258`, :psbug:`2501` (upstream :mysqlbug:`86164`).
+
+* A build errors were present on FreeBSD (caused by fixing the bug
+  :psbug:`255`) and on MacOS (caused by fixing the bug :psbug:`264`).
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: in |Percona Server| :rn:`$version` so users know which versions are affected

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+* Server queries that contained JSON special characters and were logged by
+  ``audit_log`` plugin in JSON format caused invalid output due to lack of
+  escaping. Bug fixed :psbug:`1115`.
+
+* Percona Server now uses *Travis CI*  for additional tests. Bug fixed
+  :psbug:`3777`.
+
+Other bugs fixed: :psbug:`257`, :psbug:`1090` (upstream :mysqlbug:`78048`), :psbug:`1109`, :psbug:`1127`, :psbug:`2415`,
+:psbug:`3767`, and :psbug:`3804` (upstream mysqlbug:`89598`).
+
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: only 21 entry, while milestone contains 34, some bugs (like ps-1075 and ps-3781) are missing from the list.

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+in the 5.7 series. All software developed by Percona is open-source and free.
+
+New Features
+============
+
+* A new string variable :variable:`version_suffix` allows to change suffix
+  for the |Percona Server| version string returned by the read-only
+  :variable:`version` variable. Also :variable:`version_comment` is converted
+  from a global read-only to a global read-write variable.
+
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst:25]: Missing https://jira.percona.com/browse/PS-298 description.

File: doc/source/management/data_at_rest_encryption.rst [hunk 9]
```diff
@@ -175,27 +175,28 @@ credentials stored in the credentials file. Location of this file is specified
...
-
-* ``vault_ca [optional]`` - this variable needs to be specified only when
+  will store keys.
+
+* ``token`` - a token generated by the Vault server, which ``keyring_vault``
+  will further use when connecting to the Vault. This token can be used to
+  store new keys in a secret mount point (this the only way to apply it, when
+  ``keyring_vault`` is used only for transparent data encryption, and not for
+  ``keyring_udf`` plugin). If ``keyring_udf`` plugin is combined with 
+  ``keyring_vault`` this token also allows to remove keys from the Vault, as
```
Comment (robgolebiowski) [inline doc/source/management/data_at_rest_encryption.rst]: this token also -> this token needs to allow removing . Something like this. Removing is an attribute of a token

File: doc/source/management/data_at_rest_encryption.rst [hunk 10]
```diff
@@ -231,6 +232,22 @@ System Variables
...
+.. variable:: keyring_vault_timeout
+
+  :version 5.7.21-20: Implemented
+  :cli: ``--keyring-vault-timeout``
+  :dyn: Yes
+  :scope: Global
+  :vartype: Numeric
+  :default: ``15``
+
+Range: 0 - 86400
```
Comment (hrvojem) [inline doc/source/management/data_at_rest_encryption.rst]: this should be part of the table, move to line 243 :range: 0 - 86400

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+  punctuation marks in natural language mode only, but not in boolean mode.
+  Bugs fixed :psbug:`258`, :psbug:`2501` (upstream :mysqlbug:`86164`).
+
+* Build errors were present on FreeBSD (caused by fixing the bug
+  :psbug:`255` in |Percona Server| :rn:`5.6.38-83.0`) and on MacOS (caused
+  by fixing the bug :psbug:`264` in |Percona Server| :rn:`5.7.20-19`). Bugs
+  fixed :psbug:`2284` and :psbug:`2286`.
+
+* A bunch of fixes where introduced to make |Percona Server| compile under
+  GCC 7. Bugs fixed :psbug:`3780` (upstream :mysqlbug:`89420`,
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: warning fixes (not errors, thus it was possible to compile under GCC 7 already, if needed with warnings-as-errors relaxed)

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+  :mysqlbug:`89421`, and :mysqlbug:`89422`).
+
+* CMake error took place at compilation with bundled zlib. Bug fixed
+  :psbug:`302`.
+
+* A GCC 7 warning fix introduced regression in |Percona Server| that lead to
+  a wrong SQL query built to access the remote server when Federated storage
+  engine was used. Bug fixed :psbug:`1134`.
+
+* It was possible to enable :variable:`encrypt_binlog` with no binary logging.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: with no binary or relay logging enabled

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+  Bug fixed :psbug:`287`.
+
+* Long buffer wait times where occuring on busy servers in case of several
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: IMPORT TABLESPACE, what workers?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by laurynas-biveinis** [comment]
syle -> style in the commit message

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+
+* Percona Server now uses *Travis CI*  for additional tests. Bug fixed
+  :psbug:`3777`.
+
+Other bugs fixed: :psbug:`257`, :psbug:`264`, :psbug:`1090`
+(upstream :mysqlbug:`78048`), :psbug:`1109`, :psbug:`1127`, :psbug:`2204`,
+:psbug:`2414`, :psbug:`2415`, :psbug:`3767`, :psbug:`3794`, and :psbug:`3804`
+(upstream :mysqlbug:`89598`).
+
+
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: Please add the list of CVEs that are being fixed with this release.

**Review by hrvojem — APPROVED** [review]

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 15]
```diff
@@ -0,0 +1,126 @@
+.. rn:: 5.7.21-20
+
+========================
+Percona Server 5.7.21-20
+========================
+
+Percona is glad to announce the release of Percona Server 5.7.21-20
+on February 16, 2018. Downloads are available `here
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: Date

File: doc/source/release-notes/Percona-Server-5.7.21-20.rst [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+  fixed :psbug:`2284` and :psbug:`2286`.
+
+* A bunch of fixes where introduced to remove GCC 7 compilation warnings for
+  the |Percona Server| build. Bugs fixed :psbug:`3780` (upstream
+  :mysqlbug:`89420`, :mysqlbug:`89421`, and :mysqlbug:`89422`).
+
+* CMake error took place at compilation with bundled zlib. Bug fixed
+  :psbug:`302`.
+
+* A GCC 7 warning fix introduced regression in |Percona Server| that lead to
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.21-20.rst]: s/lead/led

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
No further reviews from me are needed after addressing the comments

**Pull request merged**